### PR TITLE
Extends the Virtual DOM implementation to support Text nodes

### PR DESCRIPTION
**Pull request by johnspackman**
Extends the Virtual DOM implementation to support Text nodes, which is for use in node.js applications and when using JSX

### CODE REVIEW

**Review by cboulanger — APPROVED** [review]

File: source/class/qx/html/Root.js
Comment (oetiker) [inline source/class/qx/html/Root.js:74]: this seems to be a public method name ... why change it ?

**Review by oetiker — APPROVED** [review]
wow ... can't really judge if this will have no side effects ... but go for it, this if master and it odes look good.

please fix the tabs

File: source/class/qx/html/Root.js
Comment (johnspackman) [reply source/class/qx/html/Root.js:74]: Only to better match the meaning (it's defined in `qx.html.Node`, and can be used to "use" text nodes, so "element" is a bit of a weird name); `useElement` still exists but is deprecated, and it is rare in the existing code to need to call it so i expect the change to be low impact

**Review by hkollmann — APPROVED** [review]

**Comment by johnspackman** [comment]
> wow ... can't really judge if this will have no side effects 

Fair comment, this has been in my main repo for 9-12 months now I think, and has been in use in production for quite a while too.

Tab fixes coming up

**Review by level420 — APPROVED** [review]

File: source/class/qx/html/Root.js
Comment (oetiker) [reply source/class/qx/html/Root.js:74]: ok

**Comment by oetiker** [comment]
@derrell  ping ?

**Review by derrell — APPROVED** [review]
Blind acceptance based on others' reviews. Too much in here I don't know enough about.

**Comment by johnspackman** [comment]
thanks for reviewing; i have a bug fix i'm working on, please hold off before merging

**Comment by johnspackman** [comment]
should be ready to merge now

**Review by cboulanger — APPROVED** [review]

**Comment by rad-pat** [comment]
Seems to have had an effect on Radio & Checkboxes for me (I just tried to bring in master today)
Oh, and its not applied the Group Box header style.
Before:
image

After:
image

Is this because we have overridden some of the styles ourselves?

**Comment by johnspackman** [comment]
@rad-pat please can you create a reproducable test case and file an issue?

**Comment by rad-pat** [comment]
Yep, I'll try to put one together

**Comment by adrelino** [comment]
I have the feeling that this is the cause for worse performance when calling `setSource` repeatetly on `qx.ui.basic.Image` as used by `ListItems` in combination with material icons.
At least I just upgraded our portal to the new qooxdoo and it now hangs a lot when interacting with the GUI.
When I click on pause in the chrome debugger while the GUI hangs, I get the following stack trace:
qooxdoo_perf_stacktrace.

`qx.ui.basic.Image.js` and `qx.html.Element.js` that are at the top of the stack trace are all files changed in this commit. 

the `_willBeSeeable` function in Node.js is marked as performance hungry, which is why I guess I end up here when clicking pause
```
    /**
     * Walk up the internal children hierarchy and
     * look if one of the children is marked as root.
     *
     * This method is quite performance hungry as it
     * really walks up recursively.
     * @return {Boolean} <code>true</code> if the element will be seeable
     */
```

My question now is, why is this function called so often now after this commit?

**Comment by adrelino** [comment]
Here is a performance profile of a GUI interaction that blocked for 38 seconds! Recorded with the Performance Tab in chrome dev tools: 
performance_list_blocking
The same function names keep appearing...

**Comment by johnspackman** [comment]
Can you get this into a reproducible example? One which shows the huge hits your taking would be really useful, but if not then just a comparative example where 6.x vs master has a noticeable difference

**Comment by adrelino** [comment]
> Can you get this into a reproducible example? One which shows the huge hits your taking would be really useful, but if not then just a comparative example where 6.x vs master has a noticeable difference

I finally found the time to put a minimal example together:
```js
function makeSelectBox(){
  var box = new qx.ui.form.SelectBox();
  for (var i=0; i<50; i++) {
    box.add(new qx.ui.form.ListItem("Free text", "@MaterialIcons/text_fields/12"));
  }
  return box;
}

// Create a button
var button1 = new qx.ui.form.Button("Click me", "pull10200/test.png");

// Document is the application root
var doc = this.getRoot();

// Add button to document at fixed coordinates
doc.add(button1, {left: 100, top: 50});

// Add an event listener
button1.addListener("execute", function() {
  for(var i=0; i<3; i++){
    mainSelect.setSelection([mainSelect.getChildren()[i]]);
  }
});

var mainSelect = makeSelectBox();
doc.add(mainSelect, {left: 100, top: 100});

var vBox = new qx.ui.container.Composite(new qx.ui.layout.VBox());
doc.add(vBox, { left: 100, top: 130 });
// Add an event listener
mainSelect.addListener("changeSelection", ()=>{
  var startTime = performance.now()
  vBox.removeAll();
  /* eslint no-alert: "off" */
  for (let i = 0; i < 20; i++) {
    vBox.add(makeSelectBox());
  }
  var endTime = performance.now();
  alert(endTime - startTime);
});
```

The same code above shows a huge difference in performance between qx 6 and 7, I host both version here:
Demo qooxdoo v7: https://adrelino.github.io/pull10200
Demo qooxdoo v6: https://tinyurl.com/yc94hqv3

Just click the button or change the topmost selectbox' selection to start the measurements.

with qooxdoo v6 I get consistent timings of around 200 msec at most whenever I change the selection of the topmost selectbox.
with qooxdoo v7, the first time I get a timing of around 500msec, but which each subsequent selection change the time doubles. So in addition to beeing less performant, there seems to be a garbage collection problem or something like that.

**Pull request merged**
